pkg/clustermesh/operator/clustermesh.go

```diff
@@ -67,14 +69,20 @@ type ClusterMesh interface {
 }
 
 func newClusterMesh(lc cell.Lifecycle, params clusterMeshParams) (*clusterMesh, ClusterMesh) {
-	if params.ClusterMeshConfig == "" || !params.Cfg.ClusterMeshEnableEndpointSync {
+	if params.ClusterInfo.ID == 0 || params.ClusterMeshConfig == "" {
+		return nil, nil
+	}
+
+	if !params.Cfg.ClusterMeshEnableEndpointSync && !params.CfgMCSAPI.ClusterMeshEnableMCSAPI {
```

Comment:
This change should be reflected in the operator deployment as well:

https://github.com/cilium/cilium/blob/34055352b9253c99f64ae22ebce3e9de5ad6ab17/install/kubernetes/cilium/templates/cilium-operator/deployment.yaml#L95-L98